core/drivers/atmel_shdwc.c

```diff
@@ -178,8 +196,27 @@ static TEE_Result atmel_shdwc_probe(const void *fdt, int node,
 	return sama5d2_pm_init(fdt, shdwc_base);
 }
 
+static const struct shdwc_compat sama5d2_compat = {
+	.shdwc_as = 0,
+	.ddrc = {
+		.type_offset = AT91_DDRSDRC_MDR,
+		.type_mask = AT91_DDRSDRC_MD,
+	}
+};
+
+static const struct shdwc_compat sama7g54_compat = {
+	.shdwc_as = 1,
+};
+
 static const struct dt_device_match atmel_shdwc_match_table[] = {
-	{ .compatible = "atmel,sama5d2-shdwc" },
+	{
+		.compatible = "atmel,sama5d2-shdwc",
+		.compat_data = &sama5d2_compat
+	},
+	{
+		.compatible = "microchip,sama7g5-shdwc",
+		.compat_data = &sama7g54_compat,
```

Comment:
Minor: would `sama7g5_compat` be a better name? I mean, the "compatible" is "g5" not "g54".